src/core/server/saved_objects/service/lib/filter_utils.ts

```diff
@@ -23,7 +23,7 @@ export const validateConvertFilterToKueryNode = (
 ): KueryNode | undefined => {
   if (filter && indexMapping) {
     const filterKueryNode =
-      typeof filter === 'string' ? esKuery.fromKueryExpression(filter) : filter;
+      typeof filter === 'string' ? esKuery.fromKueryExpression(filter) : cloneDeep(filter);
```

Comment:
I discovered what looks like a bug in the `validateConvertFilterToKueryNode` that mutated the KueryNode input. This meant that if you passed a KueryNode filter to the SOR `find()` function, it could only be used once, and subsequent calls would error out with invalid expressions (because the node's `[type].attributes.[field]` expression was converted to `[type].[field]`)

Reply:
Already fixed by https://github.com/elastic/kibana/pull/97081

Reply:
Nice job on that! Great minds think alike